streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -242,11 +256,16 @@ public InternalTopicManager(final Admin adminClient, final StreamsConfig streams
                     log.error(errorMsg);
                     throw new StreamsException(errorMsg);
                 }
-            } else {
+            } else if (!leaderNotAvailableTopics.contains(topicName)) {
                 topicsToCreate.add(topicName);
             }
         }
 
         return topicsToCreate;
     }
+
+    private boolean shouldRetry(final Set<String> topicsNotReady, final HashSet<String> leaderNotAvailableTopics) {
+        // If there's topic with LeaderNotAvailableException, we still need retry
+        return !topicsNotReady.isEmpty() || leaderNotAvailableTopics.size() > 0;
```

Comment:
Can we just use `!isEmpty` for both sets?

Reply:
Good suggestion. Updated.